shm/Ship.cpp

```diff
@@ -1,20 +1,15 @@
+#include "Cargo.hpp"
 #include "Ship.hpp"
-    
-Ship::Ship() 
-    : id_(-1)
+
+Ship::Ship()
+    : Ship(0, 0, 0, "", -1, {})
 {}
 
-Ship::Ship(int capacity, int maxCrew, int speed, const std::string& name, size_t id, std::vector<std::unique_ptr<Cargo>> cargoVec)
-    : capacity_(capacity)
-    , maxCrew_(maxCrew)
-    , crew_(0)
-    , speed_(speed)
-    , name_(name)
-    , id_(id)
-    , cargoVec_(std::move(cargoVec))
+Ship::Ship(size_t capacity, size_t maxCrew, size_t speed, const std::string& name, const size_t id, CargoVec cargoVec)
+    : capacity_(capacity), maxCrew_(maxCrew), crew_(0), speed_(speed), name_(name), id_(id), cargoVec_(std::move(cargoVec))
 {}
 
-Ship::Ship(int maxCrew, int speed, size_t id)
+Ship::Ship(size_t maxCrew, size_t speed,const size_t id)
```

Comment:
Unnecesary const in id argument.
If id don't define some ship type and shouldn't be duplicated consider static id counter in class for secure different values in them.

Reply:
Good idea with a static counter. In that case you will need to handle id initialization in constructor. Something like id_ {++s_counter} in init list. Where s_counter is static variable representing last assigned id_.